TOSCA/reduce.py

```diff
@@ -0,0 +1,55 @@
+from mantid.simpleapi import *
+
+
+def generate_input_path_for_run(run_number):
+    return f"/archive/ndxtosca/Instrument/data/{cycle}/TSC{run_number}.nxs"
```

Comment:
Does this run, isn't cycle undefined. Also, I don't think cycle is available as part of a nexus file. I might be wrong though

Reply:
So I think what's happened is I am just benefiting from the fact that python uses variables that are from parent scopes even if it is not obvious, however I should refactor this a bit so we pass the cycle in.